pkg/odo/cli/storage/create.go

```diff
@@ -30,121 +25,69 @@ var (
+type CreateOptions struct {
 	storageName      string
 	storageSize      string
 	storagePath      string
 	componentContext string
 
-	devfilePath   string
-	isDevfile     bool
-	componentName string
+	storage localConfigProvider.LocalStorage
 	*genericclioptions.Context
 }
 
-// NewStorageCreateOptions creates a new StorageCreateOptions instance
-func NewStorageCreateOptions() *StorageCreateOptions {
-	return &StorageCreateOptions{devfilePath: component.DevfilePath}
+// NewStorageCreateOptions creates a new CreateOptions instance
+func NewStorageCreateOptions() *CreateOptions {
+	return &CreateOptions{}
 }
 
-// Complete completes StorageCreateOptions after they've been created
-func (o *StorageCreateOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
-	o.devfilePath = filepath.Join(o.componentContext, o.devfilePath)
-	o.isDevfile = util.CheckPathExists(o.devfilePath)
-
-	if o.isDevfile {
-		o.Context = genericclioptions.NewDevfileContext(cmd)
-		o.componentName = o.EnvSpecificInfo.GetName()
-	} else {
-		o.Context = genericclioptions.NewContext(cmd)
-		o.componentName = o.LocalConfigInfo.GetName()
+// Complete completes CreateOptions after they've been created
+func (o *CreateOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
+	o.Context, err = genericclioptions.New(genericclioptions.CreateParameters{
+		Cmd:              cmd,
+		DevfilePath:      component.DevfilePath,
+		ComponentContext: o.componentContext,
+	})
 
-		if o.storageSize == "" || o.storagePath == "" {
-			return fmt.Errorf("\"size\" and \"path\" flags are required for s2i components")
-		}
+	if err != nil {
+		return err
 	}
 
 	if len(args) != 0 {
 		o.storageName = args[0]
 	} else {
-		o.storageName = fmt.Sprintf("%s-%s", o.componentName, util.GenerateRandomString(4))
+		o.storageName = fmt.Sprintf("%s-%s", o.Context.LocalConfigProvider.GetName(), util.GenerateRandomString(4))
 	}
 
-	o.applyDevfileStorageDefault()
+	o.storage = localConfigProvider.LocalStorage{
+		Name: o.storageName,
+		Size: o.storageSize,
+		Path: o.storagePath,
+	}
 
-	return
-}
+	o.Context.LocalConfigProvider.CompleteStorage(&o.storage)
 
-func (o *StorageCreateOptions) applyDevfileStorageDefault() {
-	if o.storageSize == "" {
-		o.storageSize = adapterCommon.DefaultVolumeSize
-	}
-	if o.storagePath == "" {
-		// acc to the devfile schema, if the mount path is absent; it will be mounted at the dir with the mount name
-		o.storagePath = "/" + o.storageName
-	}
+	return
 }
 
-// Validate validates the StorageCreateOptions based on completed values
-func (o *StorageCreateOptions) Validate() (err error) {
-	if o.isDevfile {
-		return
-	}
-	// validate storage path
-	return o.LocalConfigInfo.ValidateStorage(o.storageName, o.storagePath)
+// Validate validates the CreateOptions based on completed values
+func (o *CreateOptions) Validate() (err error) {
+	// validate the storage
+	return o.LocalConfigInfo.ValidateStorage(o.storage)
```

Comment:
Why are we using the same function to validate storage for components of s2i as well as devfile? Or am I missing something?

Reply:
Fixed